cabal-install/src/Distribution/Client/VCS.hs

```diff
@@ -305,7 +314,43 @@ vcsDarcs =
 
     vcsSyncRepos :: Verbosity -> ConfiguredProgram
                  -> [(SourceRepositoryPackage f, FilePath)] -> IO [MonitorFilePath]
-    vcsSyncRepos _v _p _rs = fail "sync repo not yet supported for darcs"
+    vcsSyncRepos verbosity prog = \ case
+      [] -> pure []
+      (primaryRepo, primaryLocalDir) : secondaryRepos ->
+        [ monitorDirectoryExistence dir
+        | dir <- primaryLocalDir : (snd <$> secondaryRepos) ] <$ do
```

Comment:
I think giving this a name, or just putting it at the end of the do block could help readability a lot here.

In this case, I think desugaring the comprehension and cons would also help:
```haskell
primaryMonitor <- monitorDirectoryExistence primaryLocalDir
monitors <- traverse (monitorDirectoryExistence . snd) secondaryRepos
return $ primaryMonitor:monitors
```

Reply:
Done